modules/build/src/main/scala/scala/build/Build.scala

```diff
@@ -842,21 +844,25 @@ object Build {
             ScalacOpt(s"-Xplugin:$path")
         }
 
-        val semanticDbScalacOptions =
-          if (generateSemanticDbs)
-            if (params.scalaVersion.startsWith("2."))
-              Seq(
-                "-Yrangepos",
-                "-P:semanticdb:failures:warning",
-                "-P:semanticdb:synthetics:on",
-                s"-P:semanticdb:sourceroot:${inputs.workspace}"
-              ).map(ScalacOpt(_))
-            else
-              Seq(
-                "-Xsemanticdb",
-                "-sourceroot",
-                inputs.workspace.toString
-              ).map(ScalacOpt(_))
+        val semanticDbTargetRootOptions: Seq[ScalacOpt] =
+          (semanticDbTargetRoot match
+            case Some(targetRoot) if params.scalaVersion.startsWith("2.") =>
+              Seq(s"-P:semanticdb:targetroot:$targetRoot")
+            case Some(targetRoot) => Seq("-semanticdb-target", targetRoot.toString)
+            case None             => Nil
+          ).map(ScalacOpt(_))
+        val semanticDbScalacOptions: Seq[ScalacOpt] =
+          if generateSemanticDbs then
+            semanticDbTargetRootOptions ++ (
+              if params.scalaVersion.startsWith("2.") then
+                Seq(
+                  "-Yrangepos",
+                  "-P:semanticdb:failures:warning",
+                  "-P:semanticdb:synthetics:on",
+                  s"-P:semanticdb:sourceroot:${inputs.workspace}"
```

Comment:
Setting sourceroot, should be enough here, no? Do we need targetRoot? I don't think it's easy to set it manually since this will be somewhere deep in bloop directories

Reply:
You may notice that we have been setting the `sourceroot` value to `inputs.workspace`, I did not change that.
However, `sourceroot` does not control where SemanticDB files go, it's merely used to `Used to relativize source file paths`. as per [SemanticDB ](https://github.com/scalameta/scalameta/blob/main/semanticdb/semanticdb3/guide.md)docs.
What controls SemanticDB files is `targetroot`, which defaults to the compiler's compiled classes directory (or in other words, the `-d` compiler option).
If we need SemanticDB files to go elsewhere, I don't think there is a way to do that without passing an explicit `targetroot`.

Reply:
https://github.com/scalameta/scalameta/blob/main/semanticdb/semanticdb3/guide.md#producing-semanticdb

Reply:
I can give `sourceroot` the same treatment as `targetroot` is getting in this PR, where it could be overridden separately from `--workspace`, if that'd help.

I cannot change the `sourceroot` to "some parent directory", as because of how Scala CLI inputs may be spread, such a thing may not even exist (and we need to be internally consistent), not to mention looking for a parent directory between source directories may catch paths the user would like to isolate from his project. Remember that a project can be spread between separate drives.

The project root/workspace is thus assumed as a best approximation of what the `sourceroot` default can be.

Reply:
I don't think targetroot would help, as we would still get relativized paths the files might end up in weird places. If we set sourceroot properly, we will get proper relative paths and the default targetroot should work fine.

Reply:
I gave `sourceroot` the same treatment, so that there's a Java/Scala version agnostic way to override either or both. This should satisfy all cases, I think.